api/types/accesslist/accesslist.go

```diff
@@ -599,3 +599,78 @@ func (a *AccessList) setInitialAuditDate(clock clockwork.Clock) (err error) {
 	a.Spec.Audit.NextAuditDate, err = a.SelectNextReviewDate()
 	return trace.Wrap(err)
 }
+
+// EqualAccessListsOption is a functional option for configuring
+// the behavior of EqualAccessLists.
+type EqualAccessListsOption func(*equalAccessListsConfig)
+
+type equalAccessListsConfig struct {
+	skipClone     bool
+	resetFieldsFn func(*AccessList)
+}
+
+// WithSkipClone configures EqualAccessLists to skip cloning
+// and directly mutate the input access lists. Use this option only when you're
+// certain the input access lists can be safely modified (e.g., they're already
+// clones or will be discarded after comparison).
+func WithSkipClone() EqualAccessListsOption {
+	return func(c *equalAccessListsConfig) {
+		c.skipClone = true
+	}
+}
+
+// WithResetFieldsForReconciliation configures EqualAccessLists to reset ephemeral
+// fields before comparison. This is useful for reconciliation scenarios where you want
+// to ignore fields managed by reconcilers or the backend.
+//
+// The following fields are reset:
+//   - Metadata.Revision: Managed by the backend
+//   - Status: Contains dynamically calculated fields (member counts, assignments, etc.)
+//   - Owner.IneligibleStatus: Managed by the IneligibleStatusReconciler
+func WithResetFieldsForReconciliation() EqualAccessListsOption {
+	return func(c *equalAccessListsConfig) {
+		c.resetFieldsFn = resetFieldsForReconciliationAccessList
+	}
+}
+
+// EqualAccessLists compares two access lists for semantic equality.
+//
+// By default, this function performs a standard equality check. Use WithResetFieldsForReconciliation()
+// to ignore ephemeral fields that are managed by reconcilers or the backend, which is useful
+// for reconciliation scenarios. This function mimics the behavior of services.CompareResources
+// for AccessList types when used with WithResetFieldsForReconciliation().
+//
+// By default, this function clones the input access lists before comparison to avoid
+// modifying the originals. Use WithSkipClone() to skip cloning if the inputs can be
+// safely modified.
+func EqualAccessLists(a, b *AccessList, opts ...EqualAccessListsOption) bool {
+	cfg := equalAccessListsConfig{}
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	if !cfg.skipClone {
+		a = a.Clone()
+		b = b.Clone()
+	}
+
+	if cfg.resetFieldsFn != nil {
+		resetFieldsForReconciliationAccessList(a)
+		resetFieldsForReconciliationAccessList(b)
+	}
```

Comment:
Two things:

- there is no reason to clone if `cfg.resetFieldsFn == nil`
- there is no reason to clone both lists at all, the ephemeral values from `a` can be copied to clone of `b`